Runtime/Scripts/BaseMeshSync.Textures.cs

```diff
@@ -0,0 +1,210 @@
+                        continue;
+                    }
+
+                    var tex = mat.GetTexture(textureName);
+
+                    if (tex is RenderTexture renderTarget) {
+                        var texture = new Texture2D(renderTarget.width, renderTarget.height,
+                            UnityEngine.TextureFormat.RGBA32, true);
+                        var activeRenderTarget = RenderTexture.active;
+
+                        RenderTexture.active = renderTarget;
+                        texture.ReadPixels(new Rect(0, 0, renderTarget.width, renderTarget.height), 0, 0);
+                        texture.Apply();
+
+                        RenderTexture.active = activeRenderTarget;
+
+                        var savePath = GetSavePath(mat, textureName);
+                        TextureData.WriteToFile(savePath, texture.EncodeToPNG());
+
+                        // Setting the texture can fail if the asset database has not imported it yet.
+                        // If that happens, save it for later and try again:
+                        if (!SetSerializedTextureForMaterial(mat, textureName)) {
+                            pendingMaterialUpdates.Add(new Tuple<Material, string>(mat, textureName));
+
+                            // Set the texture on the material to null for now so the render texture
+                            // is not saved to file again if this gets called again before it was
+                            // set to the texture in the asset library:
+                            mat.SetTexture(textureName, null);
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Sets the texture on the material if it exists in the AssetDatabase, otherwise it schedules a delayed call to try again when the asset might be imported.
+        /// </summary>
+        /// <param name="mat">Material to set the texture on</param>
+        /// <param name="textureName">Name of the texture in the material</param>
+        /// <returns></returns>
+        private bool SetSerializedTextureForMaterial(Material mat, string textureName) {
+            var savePath = GetSavePath(mat, textureName);
+
+            // LoadAssetAtPath can throw an exception if it's called during a domain backup but we have no way to detect that:
+            try {
+                if (AssetDatabase.LoadAssetAtPath<Texture2D>(savePath) == null) {
+                    EditorApplication.delayCall -= UpdatePendingMaterials;
+                    EditorApplication.delayCall += UpdatePendingMaterials;
+
+                    return false;
+                }
+            }
+            catch (Exception ex) {
+                return false;
+            }
+
+            // Ensure the texture is up to date:
+            AssetDatabase.ImportAsset(savePath);
+
+            var savedTexture = AssetDatabase.LoadAssetAtPath<Texture2D>(savePath);
```

Comment:
this call to LoadAssetAtPath could also throw an exception right?

Reply:
I've refactored it to remove the second call.